srcs/app/CGI/CgiHandler.cpp

```diff
@@ -26,21 +37,217 @@ ssize_t	CgiHandler::SendCgiOutput() {
 	if (nbytes > 0) {
 		data_.erase(0, nbytes);
 	}
+	if (data_.empty()) {
+		fd_sets_->removeFd(socket_);
+	}
 	return nbytes;
 }
 
-bool	CgiHandler::HasDataAvailable() const {
-	return !data_.empty();
-}
-
 int		CgiHandler::GetSocket() const {
 	return socket_;
 }
 
 int		CgiHandler::GetCgiOutFd() const {
-	return cgi_output_;
+	return cgi_info_.cgi_output_fd;
+}
+
+bool	CgiHandler::IsComplete() const {
+	return cgi_complete_ && data_.empty();
+}
+
+void	CgiHandler::TryParseHeaders_() {
+	try {
+		ParseHeaders_();
+		if (cgi_complete_ && !headers_parsing_complete_) {
+			throw std::runtime_error("Invalid header");
+		}
+	}
+	catch (const std::exception &) {
+		fd_sets_->removeFd(cgi_info_.cgi_output_fd);
+		SetErrorResponse_(500);
+		if (!cgi_complete_) {
+			kill(cgi_info_.pid, SIGTERM);
```

Comment:
Why SIGTERM?
I'm a fan of SIGKILL, but maybe I don't have any particular reason other than the fact that the child could not ignore it or otherwise handle it.

Reply:
Agree. I think SIGKILL would be better in this case.